File: Modulo 1/Clase06/app/src/main/java/com/area51/clase07/MainActivity.java
package com.area51.clase07;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import com.area51.clase07.sqlite.MetodoSQLite;

public class MainActivity extends AppCompatActivity {
    private EditText etUsuario;
    private EditText etContrasenia;
    private Button btnIniciar;
    private TextView tvRegistrar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        etUsuario = findViewById(R.id.etUsuario);
        etContrasenia = findViewById(R.id.etContrasena);
        btnIniciar = findViewById(R.id.btnIniciar);
        tvRegistrar = findViewById(R.id.tvRegistrar);
    }

    @Override
    protected void onResume() {
        super.onResume();

        btnIniciar.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                String usuario = etUsuario.getText().toString();
                String contrasenia = etContrasenia.getText().toString();

                MetodoSQLite sqLite = new MetodoSQLite(MainActivity.this);
                Usuario datos = sqLite.validarUsuario(usuario, contrasenia);
                if (datos == null) {
                    Toast.makeText(MainActivity.this,
                            "Usuario y/o contraseña incorrectos",
                            Toast.LENGTH_SHORT).show();
                } else {
                    Intent intent = new Intent(
                            MainActivity.this,
                            MenuActivity.class);
                    intent.putExtra("item", datos);
                    startActivity(intent);
                }
            }
        });

        tvRegistrar.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent =
                        new Intent(MainActivity.this,
                                RegistroActivity.class);
                startActivity(intent);
            }
        });
    }
}

File: Modulo 3/ClaseMVP/app/src/main/java/com/area51/clasemvp/ui/login/activity/MainActivity.kt
package com.area51.clasemvp.ui.login.activity

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import com.area51.clasemvp.R
import com.area51.clasemvp.net.entidad.Usuario
import com.area51.clasemvp.ui.login.presenter.LoginPresenter
import com.area51.clasemvp.ui.login.views.LoginView
import com.area51.clasemvp.ui.registro.activity.RegistroActivity
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity(), LoginView {
    override fun respuestaValidarUsuario(status: Boolean, usuario: Usuario?) {
        if (status) {
            Toast.makeText(this,
                    "Usuario correcto", Toast.LENGTH_SHORT).show()
        } else {
            Toast.makeText(this,
                    "Usuario incorrecto", Toast.LENGTH_SHORT).show()
        }
    }

    override fun irPantallaRegistro() {
        val intent = Intent(this, RegistroActivity::class.java)
        startActivity(intent)
    }

    var presenter: LoginPresenter? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        presenter = LoginPresenter()
        presenter!!.agregarView(this)

        //presenter!!.cargarDatos()
    }

    override fun onDestroy() {
        super.onDestroy()

        if (presenter != null)
            presenter!!.eliminarView()
    }

    override fun onResume() {
        super.onResume()

        btnIngresar.setOnClickListener {
            val usuario = etUsuario.text.toString()
            val contrasena = etContrasena.text.toString()

            presenter!!.validarUsuario(usuario, contrasena)
        }
        tvRegistrar.setOnClickListener {
            presenter!!.irARegistro()
        }
    }
}
